HypeBears.sol

```diff
@@ -0,0 +1,182 @@
+        uint256 amount1 = balance * 70 / 100;
+        uint256 amount2 = balance - amount1;
+        payable(0xe0F7204f04b060715f858Ba8Ae357f57E5494d18).transfer(amount1);
+        payable(0x029c2D9EDC080A5A077f30F3bf6122e100F2aDc6).transfer(amount2);
+    }
+
+    function updateMintLimit(uint256 _newLimit) public onlyOwner {
+        mintLimit = _newLimit;
+    }
+
+    function updateMintPrice(uint256 _newPrice) public onlyOwner {
+        mintPrice = _newPrice;
+    }
+
+    function addPartnerMint(address account, uint256 amount) public onlyOwner {
+        partnerMintAmount += amount;
+        require(totalSupply().add(partnerMintAmount) <= TOTAL_NFT, "Can't add partner more than available");
+        partnerMintAvailableBy[account] += amount;
+    }
+
+    function mintNFT(uint256 _numOfTokens, bytes memory _signature) public payable {
+        require(mintActive, 'Not active');
+        require(_numOfTokens <= mintLimit, "Can't mint more than limit per tx");
+        require(mintPrice.mul(_numOfTokens) <= msg.value, "Insufficient payable value");
+        require(totalSupply().add(_numOfTokens).add(partnerMintAmount) <= TOTAL_NFT, "Can't mint more than 10000");
+        (bool success, string memory reason) = canMint(msg.sender, _signature);
+        require(success, reason);
+
+        for(uint i = 0; i < _numOfTokens; i++) {
+            _safeMint(msg.sender, totalSupply() + 1);
+        }
+        addressMinted[msg.sender] = true;
+    }
+
+    function partnersMintMultiple(address[] memory _to) public {
+        uint256 amount = _to.length;
+        require(partnerMintAmount >= amount, "Can't mint more than total available for partners");
+        require(partnerMintAvailableBy[msg.sender] >= amount, "Can't mint more than available for msg.sender");
+        for(uint256 i = 0; i < amount; i++){
+            _safeMint(_to[i],totalSupply() + 1);
+        }
+        partnerMintAmount -= amount;
+        partnerMintAvailableBy[msg.sender] -= amount;
+    }
+
+    function tokenURI(uint256 _tokenId) public view virtual override returns (string memory) {
+        require(_exists(_tokenId), "ERC721Metadata: URI query for nonexistent token");
+        if (!reveal) {
+            return string(abi.encodePacked(blindURI));
+        } else {
+            return string(abi.encodePacked(baseURI, _tokenId.toString()));
+        }
+    }
+
+    function supportsInterface(bytes4 _interfaceId) public view override (ERC721, ERC721Enumerable) returns (bool) {
+        return super.supportsInterface(_interfaceId);
+    }
+
+    function _beforeTokenTransfer(address _from, address _to, uint256 _tokenId) internal override(ERC721, ERC721Enumerable) {
+        super._beforeTokenTransfer(_from, _to, _tokenId);
+    }
+
+    function isApprovedForAll(address owner, address operator) override public view returns(bool) {
+        // Whitelist OpenSea proxy contract for easy trading.
+        if (proxyRegistryAddress == operator) {
```

Comment:
I don't believe this correctly queries the proxy.

See OpenSea example [here](https://github.com/ProjectOpenSea/opensea-creatures/blob/master/contracts/ERC721Tradable.sol#L85)